### PR TITLE
feat: add capability of external dns services

### PR DESCRIPTION
**Pull request by smuu**
Resolves https://github.com/celestiaorg/knuu/issues/120

Tested here: https://github.com/celestiaorg/knuu-example/blob/8439f2ba8104b84464a48ca57fa6c734e05fc6b8/basic/external-dns_test.go

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: pkg/k8s/k8s_service.go [hunk 1]
```diff
@@ -28,9 +28,9 @@ func GetService(namespace, name string) (*v1.Service, error) {
 }
 
 // DeployService deploys a service if it does not exist.
-func DeployService(namespace, name string, labels, selectorMap map[string]string, portsTCP []int, portsUDP []int) (*v1.Service, error) {
+func DeployService(namespace, name string, labels, annotations, selectorMap map[string]string, portsTCP []int, portsUDP []int, serviceType v1.ServiceType) (*v1.Service, error) {
```
Comment (jrmanes) [inline pkg/k8s/k8s_service.go:31]: as this function is growing, what do you think if for a better readability we change the style to:

```go
func DeployService(
	namespace, 
	name string, 
	labels, 
	annotations, 
	selectorMap map[string]string, 
	portsTCP []int, 
	portsUDP []int, 
	serviceType v1.ServiceType) 
(*v1.Service, error) { 
```

File: pkg/k8s/k8s_service.go [hunk 2]
```diff
@@ -126,8 +126,8 @@ func buildPorts(tcpPorts, udpPorts []int) []v1.ServicePort {
 }
 
 // prepareService constructs a new Service object with the specified parameters.
-func prepareService(namespace, name string, labels, selectorMap map[string]string,
-	tcpPorts, udpPorts []int) (*v1.Service, error) {
+func prepareService(namespace, name string, labels, annotations, selectorMap map[string]string,
```
Comment (jrmanes) [inline pkg/k8s/k8s_service.go:129]: same to this one, using the previous style

File: pkg/knuu/instance.go [hunk 3]
```diff
@@ -609,6 +610,22 @@ func (i *Instance) SetStartupProbe(startupProbe *v1.Probe) error {
...
+	}
+	// check if external DNS is already added
+	for _, d := range i.externalDns {
+		if d == dns {
+			return fmt.Errorf("external DNS '%s' is already added", dns)
+		}
+	}
+	i.externalDns = append(i.externalDns, dns)
+	return nil
+}
```
Comment (Bidon15) [inline pkg/knuu/instance.go:627]: Our of curiosity, is it planned to have removeExternalDNS in the future?

File: pkg/k8s/k8s_service.go (hunk 1 above)
Comment (smuu) [reply pkg/k8s/k8s_service.go]: https://github.com/celestiaorg/knuu/pull/152/commits/e371faab7c9f662bef4d180d9ad496c4a8a6a60f

File: pkg/k8s/k8s_service.go (hunk 2 above)
Comment (smuu) [reply pkg/k8s/k8s_service.go:144]: https://github.com/celestiaorg/knuu/pull/152/commits/e371faab7c9f662bef4d180d9ad496c4a8a6a60f

File: pkg/knuu/instance.go (hunk 3 above)
Comment (smuu) [reply pkg/knuu/instance.go:628]: I don't see the use case right now. I could add it for completion tho.

Comment (smuu) [reply pkg/knuu/instance.go:628]: Agreed with @jrmanes that knuu will not alter an existing environment but rather deploy a new one.

**Pull request merged**